core/editor/src/dispatcher/mod.rs

```diff
@@ -9,83 +9,74 @@ pub struct Dispatcher {
 }
 
 impl Dispatcher {
-	pub fn handle_event(&self, state: &mut EditorState, event: Event) -> Result<(), EditorError> {
+	pub fn handle_event(&self, tool_state: &mut ToolState, document: &mut Document, event: &Event) -> Result<(), EditorError> {
```

Comment:
why is this necessary? passing the whole editor state seems more natural to me and is also more extensible.

Reply:
@geom3trik was this resolved yet? Feel free to hit "Resolve Conversation" if the button exists for you.